### PR TITLE
WINC-1035: Add the nutanix-e2e-operator WMCO test job

### PR DESCRIPTION
**Pull request by yanhua121**
Add the nutanix-e2e-operator job to the WMCO CI tests.

### CODE REVIEW

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for working on this, @yanhua121. Please address my comments.

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml [hunk 1]
```diff
@@ -196,3 +196,11 @@ zz_generated_metadata:
   branch: master
   org: openshift
   repo: windows-machine-config-operator
+- as: nutanix-e2e-operator
+  optional: true
+  skip_if_only_changed: ^(?:docs|\.github)/|\.md$|^(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
```
[the remark this answers is not in the document]
Comment (saifshaikh48) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml:201]: @yanhua121 `always_run: false` is what you're likely looking for, [example](https://github.com/openshift/release/pull/37634/commits/4dd4c2cc18e94ed185bff313e6c90cd13c1fdfce#diff-8e8dbaad2b4e85436d575ecf47299ac4cdae287329bd4533fa535b2641fd6ff5R158)

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml [hunk 2]
```diff
@@ -196,3 +196,11 @@ zz_generated_metadata:
...
+  steps:
+    cluster_profile: nutanix
+    test:
+    - ref: windows-e2e-operator-test
+    workflow: openshift-e2e-nutanix-ovn
```
[the remark this answers is not in the document]
Comment (yanhua121) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml]: Done

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (yanhua121) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml]: Fixed

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml [hunk 3]
```diff
@@ -196,3 +196,11 @@ zz_generated_metadata:
   branch: master
```
[the remark this answers is not in the document]
Comment (yanhua121) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml]: Added the generated job by running "make jobs".

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml [hunk 4]
```diff
@@ -192,6 +192,14 @@ tests:
...
       SOURCE_BRANCH: master
     workflow: windows-ci-fastforward
+- as: nutanix-e2e-operator
+  always_run: false
+  optional: true
+  steps:
+    cluster_profile: nutanix
+    test:
+    - ref: windows-e2e-operator-test
+    workflow: openshift-e2e-nutanix-ovn
```
Comment (mansikulkarni96) [inline ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml:202]: This workflow does not have a ovn-hybird-manifest step. Do we need to add that?

Comment (aravindhp) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml:202]: Yes we will need that for Windows support. Ref: https://docs.openshift.com/container-platform/4.12/networking/ovn_kubernetes_network_provider/configuring-hybrid-networking.html

**Comment by aravindhp** [comment]
/pj-rehearse pull-ci-openshift-windows-machine-config-operator-master-nutanix-e2e-operator

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml [hunk 5]
```diff
@@ -192,6 +192,14 @@ tests:
       DESTINATION_BRANCH: community-4.14
       SOURCE_BRANCH: master
     workflow: windows-ci-fastforward
+- always_run: false
+  as: nutanix-e2e-operator
```
Comment (adiantum) [inline ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml:196]: Add cluster: build01 like here: https://github.com/openshift/release/blob/81652219f121ff71071a9ec7d495fb8a827247c4/ci-operator/config/openshift/installer/openshift-installer-release-4.13.yaml#L615

File: ci-operator/step-registry/ipi/nutanix/ovn/hybrid/ipi-nutanix-ovn-hybrid-workflow.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+workflow:
+  as: ipi-nutanix-ovn-hybrid
+  steps:
+    pre:
+    - chain: ipi-conf-nutanix
+    - ref: ovn-conf
+    - ref: ovn-conf-hybrid-manifest
+    - chain: ipi-install
+    post:
+    - chain: ipi-deprovision
```
Comment (adiantum) [inline ci-operator/step-registry/ipi/nutanix/ovn/hybrid/ipi-nutanix-ovn-hybrid-workflow.yaml:10]: Take a look https://github.com/openshift/release/blob/81652219f121ff71071a9ec7d495fb8a827247c4/ci-operator/step-registry/ipi/nutanix/post/ipi-nutanix-post-chain.yaml#L4

Comment (yanhua121) [reply ci-operator/step-registry/ipi/nutanix/ovn/hybrid/ipi-nutanix-ovn-hybrid-workflow.yaml]: changed to use "ipi-nutanix-post".

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml (hunk 5 above)
Comment (yanhua121) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml:196]: Fixed.

**Comment by yanhua121** [comment]
/pj-rehearse pull-ci-openshift-windows-machine-config-operator-master-nutanix-e2e-operator

**Review by aravindhp — APPROVED** [review]

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

**Comment by mtnbikenc** [comment]
/retitle WINC-1035 Add the nutanix-e2e-operator WMCO test job

**Comment by mtnbikenc** [comment]
/retitle WINC-1035: Add the nutanix-e2e-operator WMCO test job

**Pull request merged**